core/math/a_star.cpp

```diff
@@ -62,6 +62,15 @@ void AStar::add_point(int p_id, const Vector3 &p_pos, real_t p_weight_scale) {
 		points[p_id]->pos = p_pos;
 		points[p_id]->weight_scale = p_weight_scale;
 	}
+
+	// calculate our new next available point id if bigger than before or next id already contained in set of points.
+	if (p_id > last_max_id || points.has(last_max_id + 1)) {
+		int cur_new_id = last_max_id;
+		while (points.has(cur_new_id + 1)) {
+			cur_new_id++;
+		}
+		last_max_id = cur_new_id;
+	}
```

Comment:
Not sure about this change. `add_point` is likely to be used more than `get_available_point_id`, especially in larger projects where performance is needed. Can the loop be moved to `get_available_point_id` somehow?

Reply:
Moved it to get_available_point_id